tests/drivers/can/api/testcase.yaml

```diff
@@ -18,3 +16,13 @@ tests:
       - esp32s2_saola
       - esp32s3_devkitm
       - xiao_esp32s3
+  drivers.can.api.nxp_s32_canxl_non_rx_fifo:
+    extra_configs:
+      - CONFIG_CAN_NXP_S32_RX_FIFO=n
+      - CONFIG_CAN_FD_MODE=n
+    filter: dt_compat_enabled("nxp,s32-canxl")
+    platform_allow:
+      - s32z270dc2_rtu0_r52
+      - s32z270dc2_rtu1_r52
+      - s32z270dc2_rtu0_r52@D
+      - s32z270dc2_rtu1_r52@D
```

Comment:
Why both a `filter` and `platform_allow`?

Reply:
I think need both them to make sure to have a dts canxl node is enabled

Reply:
yes, it is not needed to use both. I removed `platform_allow`